include/nbl/ui/IInputEventChannel.h

```diff
@@ -152,13 +160,19 @@ class IMouseEventChannel : public impl::IEventChannelBase<SMouseEvent>
 
 struct SKeyboardEvent
 {
+    SKeyboardEvent() {
+        using namespace std::chrono;
+        timeStamp = duration_cast<microseconds>(system_clock::now().time_since_epoch());
+    }
```

Comment:
this constructor and timeStamp member variable repeats a lot, have you thought about making a common `SEvent` and having the other structs inherit from it?

Reply:
also this is not when the time should be polled!